system/netdata.service.in

```diff
@@ -1,15 +1,56 @@
+# netdata systemd target
+
 [Unit]
-Description=Real time performance monitoring
+Description=netdata - Real-time performance monitoring
+Documentation=man:netdata
+Documentation=file:///usr/share/doc/netdata/html/index.html
+Documentation=https://github.com/firehol/netdata
 After=network.target httpd.service squid.service nfs-server.service mysqld.service named.service postfix.service
+Wants=network-online.target
+ConditionPathExists=/etc/netdata/netdata.conf
 
 [Service]
 Type=simple
+Environment="netdata_LOG_LOCATION=/var/log/netdata/log"
+ExecStart=@sbindir_POST@/netdata -D
+ExecReload=@sbindir_POST@/netdata reload
+TimeoutStopSec=60
+KillMode=mixed
+KillSignal=SIGTERM
+OOMScoreAdjust=-900
+
 User=netdata
 Group=netdata
-ExecStart=@sbindir_POST@/netdata -D
+Restart=on-abnormal
+RestartSec=2s
```

Comment:
The counter starts counting when daemon has been terminated (to prevent crazy restart loops).
Systemd is not monitoring if Netdata is doing its things at runtime or frozen/stuck in a loop without exiting.

However, there is a nice watchdog feature where the daemon would have to "ping" the service manager every X seconds to prove to be healthy [1]. Failure to ping would lead to a restart.

[1] https://www.freedesktop.org/software/systemd/man/sd_notify.html